rpcq/_base.py

```diff
@@ -100,10 +69,7 @@ def astuple(self):
         :return: A tuple representation of the message.
         :rtype: Tuple[Any]
         """
-        raise NotImplementedError(self.__class__.__name__)
-
-    def __hash__(self):
-        return hash((self.__class__, self.astuple()))
+        return tuple(self.__dict__.values())
```

Comment:
Python 3 dictionaries maintain the insertion order, but I would nonetheless prefer making explicit in which order field values are stored.

Reply:
Why do we need to implement `asdict` `astuple` anyway? It seems like Dataclass already contains those...

Reply:
The python message generator has to sort the slots in a message, including some of the messages in our internal extension, because a dataclass initialization function is required to have required arguments come before optional ones. What sort of sorting are you expecting `astuple` to come up with? Something consistent with the message spec definition ordering or something you could pass to `__init__`, with whatever sorting the generator comes up with?

Reply:
I'm happy to drop asdict and astuple if we can show they're not being used

Reply:
we can drop `astuple` but `asdict` is used in a few places in `pidgin`

Reply:
That's fine: I don't think `asdict` is such a sin. I'll drop `astuple` and see if anything breaks.